### PR TITLE
Migrate the ANU hub to use the JupyterHub CILogonOAuthenticator

### PR DESCRIPTION
**Pull request by GeorgianaElena**
Part of https://github.com/2i2c-org/infrastructure/issues/967.
Hopefully, this shouldn't disrupt the current users too much, and usernames should be preserved.

### CODE REVIEW

**Comment by sgibson91** [comment]
@GeorgianaElena I believe you also need to add `enc-anu.secret.values.yaml` to the helm chart values files list in `cluster.yaml`

https://github.com/2i2c-org/infrastructure/blob/ec7122536dddc9636833831be24ba02f72e79bfc/config/clusters/2i2c/cluster.yaml#L150-L154

File: config/clusters/2i2c/anu.values.yaml [hunk 1]
```diff
@@ -30,9 +30,14 @@ jupyterhub:
...
+        username_claim: "email"
         admin_users:
           - matthew.mckay@anu.edu.au
-        # We do not define allowed_users here since only usernames matching this regex will be allowed to login into the hub.
-        # Ref: https://jupyterhub.readthedocs.io/en/stable/api/auth.html#jupyterhub.auth.Authenticator.username_pattern
-        username_pattern: '^(.+@2i2c\.org|.+@anu\.edu\.au|deployment-service-check)$'
+        # We do not define allowed_users here since only usernames matching these domains will be allowed to login into the hub.
+        allowed_idps:
+          - 2i2c.org
+          - anu.edu.au
```
[the remark this answers is not in the document]
Comment (yuvipanda) [reply config/clusters/2i2c/anu.values.yaml:43]: Don't worry, we're working on ways to make it harder to read ;)

**Review by yuvipanda — APPROVED** [review]

**Comment by mmcky** [comment]
thanks @choldgraf in the tag. Is this going to change how the interface looks when logging in? I may need to prepare something for the class which is halfway through semester so students don't get too confused with a sudden change.

**Comment by GeorgianaElena** [comment]
@sgibson91, the cilogon credentials secret file is listed in the `helm_chart_values_files`.

https://github.com/2i2c-org/infrastructure/blob/0e1294773a4d2ee0d7f208f041b7156e204c1dad/config/clusters/2i2c/cluster.yaml#L150-L155

Btw, I missed the change too when reviewing Yuvi's PR here https://github.com/GeorgianaElena/pilot-hubs/pull/3#issuecomment-1077905775. I don;t know how, but it has great camouflage powers 😅

**Comment by GeorgianaElena** [comment]
> Is this going to change how the interface looks when logging in?

@mmcky, the interface doesn't change too much. The only difference it's that you won't have to click the "Sign in with CILogon" button below ⬇️ but will be presented with the CILogon login screen directly. 
auth0-cilogon-button 

The demo hub https://demo.2i2c.cloud already has this version deployed if you want to try it out.

**Comment by sgibson91** [comment]
> I don;t know how, but it has great camouflage powers 😅

_Great_ camouflage!

**Comment by mmcky** [comment]
Oh I see -- thanks @GeorgianaElena. 👍

**Pull request merged**